### ISSUE TITLE
6157 - UiThreadingViolationException when switching L&F

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#2213]
No noticeable impact, but after switching L&F the following stack trace is in the console:

```
org.pushingpixels.substance.api.UiThreadingViolationException: Component creation must be done on Event Dispatch Thread
	at org.pushingpixels.substance.internal.utils.SubstanceCoreUtilities.testComponentCreationThreadingViolation(SubstanceCoreUtilities.java:2048)
	at org.pushingpixels.substance.internal.ui.SubstanceRootPaneUI.createUI(SubstanceRootPaneUI.java:207)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at sun.reflect.misc.Trampoline.invoke(MethodUtil.java:71)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at sun.reflect.misc.MethodUtil.invoke(MethodUtil.java:275)
	at javax.swing.UIDefaults.getUI(UIDefaults.java:769)
	at javax.swing.UIManager.getUI(UIManager.java:1016)
	at javax.swing.JRootPane.updateUI(JRootPane.java:484)
	at javax.swing.JRootPane.<init>(JRootPane.java:371)
	at javax.swing.JFrame.createRootPane(JFrame.java:286)
	at javax.swing.JFrame.frameInit(JFrame.java:267)
	at javax.swing.JFrame.<init>(JFrame.java:234)
	at tools.map.making.MapCreator.<init>(MapCreator.java:85)
	at tools.map.making.MapCreator.main(MapCreator.java:78)
```

**Comment by RoiEXLab** [comment]
The important question is why this code is being executed by the map creator

**Comment by ssoloff** [comment]
I was able to reproduce the problem by running Map Creator from the main menu with the Substance L&F active.  It doesn't matter if you do it immediately after changing the L&F or after a full restart of TripleA.  Any time you start Map Creator, a bunch of similar exceptions are reported.

A quick examination of the `MapCreator` code shows that a lot of Swing-related stuff is being done on the main thread instead of the EDT, so there's definitely a problem there.  The Substance L&F is probably a bit more aggressive about ensuring stuff happens on the EDT than the Sun/Oracle L&Fs.

**Comment by ron-murhammer** [comment]
@ssoloff @DanVanAtta So does the Map Creator not work with the standard L&F Substance Graphite without errors? Point being if we make that the default and very few people change it and not the many folks use the map creator that number of people would be low.

**Comment by ssoloff** [comment]
@ron-murhammer I believe it still works.  It simply is reporting what are the equivalent of precondition violations when the UI is initially created in `main()`.  As I mentioned above, the problem is there regardless of which L&F is used; it just happens to be that Substance has installed a handler to ensure all UI operations occur on the EDT.

I got a bit of déjà vu when trying to reproduce the problem.  One of the first bugs I fixed was in the Map Creator, and I was running Substance at the time.  I seem to remember this flury of exceptions in the console when I ran the Map Creator back then, so I suspect the problem has been there at least since 1.9.0.0.3635 was released, if not earlier.
